Assets/World/Units/Unit.gd

```diff
@@ -31,29 +31,31 @@ const DIRECTION = [
 	Vector2( 1, -1), # UP
 	Vector2( 1,  0)] # UP_RIGHT
 
-#warning-ignore-all:unused_class_variable
-
 # Generic properties
 @export var unit_name = "Untitled" # user defined name for the unit
-@export var faction: Global.Faction = 0 : set = set_faction
-var health = -1 # health must be set or it won't auto destroy itself
+@export var faction: Global.Faction = 0:
+	set(new_faction):
+		faction = new_faction
+
+var health := -1 # health must be set or it won't auto destroy itself
 
 # Pathfinding
 var path = []
-var path_index = 0
+var path_index := 0
 
 var move_vector = Vector2()
 
 # Sprite2D rotation
-var direction = -1 # for non-animated movements, this is the frame_index
-var rotation_offset = 0
+var direction := -1 # for non-animated movements, this is the frame_index
+var rotation_offset := 0
 var rotation_index
 
 var buoy: Buoy = null
 
-@onready var rotation_y: Node3D = get_node_or_null("/root/World/PlayerCamera/RotationY") as Node3D
-@onready var _as_map: Node3D = get_node_or_null("/root/World/AStarMap") as Node3D
-@onready var world: Node3D = get_node_or_null("/root/World") as Node3D
+@onready var world := get_node_or_null("/root/World") as Node3D
+
+@onready var _as_map := get_node_or_null("/root/World/AStarMap") as AStarMap
+@onready var _rotation_y := get_node_or_null("/root/World/PlayerCamera/RotationY") as Node3D
```

Comment:
If you want to do it like that, you should verify if the node is not null somewhere